chainerx/_fallback_workarounds.py

```diff
@@ -0,0 +1,143 @@
+# This file defines workaround implementation for
+# NumPy-compatibility functions that fall back to NumPy/CuPy functions
+# for native/cuda devices respecitvely.
+# The workaround does not support backprop, and also requires external
+# libraries mentioned above.
+# Functions defined in this file should be considered to have high priority for
+# genuine implementations.
+
+import numpy
+
+import chainerx
+
+
+try:
+    import cupy
+except Exception:
+    cupy = None
+
+
+def _to_numpy(array):
+    assert isinstance(array, chainerx.ndarray)
+    return chainerx.to_numpy(array, copy=False)
+
+
+def _from_numpy(array):
+    assert isinstance(array, numpy.ndarray)
+    return chainerx.array(array, copy=False)
+
+
+def _to_cupy(array):
+    # Convert to cupy.ndarray on the same device as source array
+    if cupy is None:
+        raise RuntimeError(
+            'Currently cupy is required in this operation.')
+    return cupy.ndarray(
+        array.shape,
+        array.dtype,
+        cupy.cuda.MemoryPointer(
+            cupy.cuda.UnownedMemory(
+                array.data_ptr + array.offset,
+                array.data_size,
+                array,
+                array.device.index),
+            0),
+        strides=array.strides)
+
+
+def _from_cupy(array):
+    if cupy is None:
+        raise RuntimeError(
+            'Currently cupy is required in this operation.')
```

Comment:
A detail but you can assert that `cupy is not None`. Maybe this error should be thrown by the caller. That would be friendlier than "ChainerX fallback implementation only supports native or cuda backends.", in case cupy isn't available.

Reply:
Okay, let's leave it here as is if it was meant to be reused. It is currently only called from `_to_chainerx` (which I mean by "caller", sorry if it was unclear) and this code is never executed so that's why I asked.

Reply:
Ah, I see. Then I think that's better.

Reply:
Fixed as suggested.